src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowTypeController.cs

```diff
@@ -475,8 +475,8 @@ public async Task<IActionResult> Edit(WorkflowTypeUpdateModel model)
             foreach (var activityState in activities)
             {
                 var activity = currentActivities[activityState["id"].ToString()];
-                activity.X = (int)Convert.ToDecimal(activityState["x"].ToString());
-                activity.Y = (int)Convert.ToDecimal(activityState["y"].ToString());
+                activity.X = activityState["x"].Value<int>();
+                activity.Y = activityState["y"].Value<int>();
```

Comment:
Alternative solution, which works standalone (without the rest of the changes):
```suggestion
                activity.X = (int)Math.Round(Convert.ToDecimal(activityState["x"].ToString(), CultureInfo.InvariantCulture), 0);
                activity.Y = (int)Math.Round(Convert.ToDecimal(activityState["y"].ToString(), CultureInfo.InvariantCulture), 0);
```

Reply:
I totally agree. Removed the script changes.